client-tez/tez/src/main/java/org/apache/tez/dag/app/CelebornDagAppMaster.java

```diff
@@ -0,0 +1,358 @@
+
+      // disable tez slow start
+      Configuration conf = new Configuration(new YarnConfiguration());
+      conf.setFloat(ShuffleVertexManager.TEZ_SHUFFLE_VERTEX_MANAGER_MIN_SRC_FRACTION, 1.0f);
+      conf.setFloat(ShuffleVertexManager.TEZ_SHUFFLE_VERTEX_MANAGER_MAX_SRC_FRACTION, 1.0f);
+      // disable reschedule task on unhealthy nodes because shuffle data are stored in Celeborn
+      conf.setBoolean(TEZ_AM_NODE_UNHEALTHY_RESCHEDULE_TASKS, false);
+
+      // support set celeborn master endpoints from env
+      String masterEndpointsKey = TezUtils.TEZ_PREFIX + CelebornConf.MASTER_ENDPOINTS().key();
+      String masterEndpointsVal = conf.get(masterEndpointsKey);
+      if (masterEndpointsVal == null || masterEndpointsVal.isEmpty()) {
+        Logger.info(
+            "MRAppMaster sets {} via environment variable {}.",
+            masterEndpointsKey,
+            MASTER_ENDPOINTS_ENV);
+        conf.set(masterEndpointsKey, TezUtils.ensureGetSysEnv(MASTER_ENDPOINTS_ENV));
+      }
+
+      DAGProtos.ConfigurationProto confProto =
+          TezUtilsInternal.readUserSpecifiedTezConfiguration(
+              System.getenv(ApplicationConstants.Environment.PWD.name()));
+      TezUtilsInternal.addUserSpecifiedTezConfiguration(conf, confProto.getConfKeyValuesList());
+
+      DAGProtos.AMPluginDescriptorProto amPluginDescriptorProto = null;
+      if (confProto.hasAmPluginDescriptor()) {
+        amPluginDescriptorProto = confProto.getAmPluginDescriptor();
+      }
+
+      UserGroupInformation.setConfiguration(conf);
+      Credentials credentials = UserGroupInformation.getCurrentUser().getCredentials();
+
+      TezUtilsInternal.setSecurityUtilConfigration(Logger, conf);
+
+      CelebornDagAppMaster appMaster =
+          new CelebornDagAppMaster(
+              applicationAttemptId,
+              containerId,
+              nodeHostString,
+              Integer.parseInt(nodePortString),
+              Integer.parseInt(nodeHttpPortString),
+              new SystemClock(),
+              appSubmitTime,
+              sessionModeCliOption,
+              System.getenv(ApplicationConstants.Environment.PWD.name()),
+              TezCommonUtils.getTrimmedStrings(
+                  System.getenv(ApplicationConstants.Environment.LOCAL_DIRS.name())),
+              TezCommonUtils.getTrimmedStrings(
+                  System.getenv(ApplicationConstants.Environment.LOG_DIRS.name())),
+              clientVersion,
+              credentials,
+              jobUserName,
+              amPluginDescriptorProto);
+      ShutdownHookManager.get()
+          .addShutdownHook(new DAGAppMasterShutdownHook(appMaster), SHUTDOWN_HOOK_PRIORITY);
+
+      // log the system properties
+      if (Logger.isInfoEnabled()) {
+        String systemPropsToLog = TezCommonUtils.getSystemPropertiesToLog(conf);
+        if (systemPropsToLog != null) {
+          Logger.info(systemPropsToLog);
+        }
+      }
+
```

Comment:
There are some succeed tasks which may be recomputed in local shuffle, for example, node manager is unhealthy, reducer could not fetch the shuffle data from mapper which has already run successfully in unhealthy node manager. On the contrary, the shuffle data is stored in celeborn, which causes that recomputing the succeeded task which have run in unhealthy node is unnecessary. Meanwhile, `TezConfiguration.TEZ_AM_NODE_UNHEALTHY_RESCHEDULE_TASKS` could set to false for this situation.

Reply:
I have done this.  Check the code https://github.com/apache/celeborn/blob/3c15db953d57a017b4e4fdc4c3cfe1956c48ae42/client-tez/tez/src/main/java/org/apache/tez/dag/app/CelebornDagAppMaster.java#L292-L294